File: .agent/templates/.docker/java-spring/Dockerfile.java
# ==========================================
# ESTÁGIO 1: BASE E DEPENDÊNCIAS (CACHE)
# ==========================================
FROM eclipse-temurin:21-jdk-alpine AS base

LABEL maintainer="Ambiente java spring boot"
LABEL description="Ambiente de desenvolvimento Java com Spring Boot"

WORKDIR /app

# Adicione pacotes base necessários

# ==========================================
# ESTÁGIO 2: DESENVOLVIMENTO (Docker)
# ==========================================
FROM base AS dev

EXPOSE 8080

# O código é provido via Bind Mount, o spring-boot run cuida do reload (spring-boot-devtools)
CMD ["sh", "-c", "./mvnw spring-boot:run -Dspring-boot.run.jvmArguments='-Dspring.devtools.restart.enabled=true'"]

# ==========================================
# ESTÁGIO 3: PRODUÇÃO (Deploy)
# ==========================================
FROM base AS prod

COPY . .

# Comando de build de produção 
RUN ./mvnw clean package -DskipTests

EXPOSE 8080

CMD ["java", "-jar", "target/app.jar"]

File: .agents/templates/.docker/go/Dockerfile.go
# ==========================================
# ESTÁGIO 1: BASE E DEPENDÊNCIAS (CACHE)
# ==========================================
FROM golang:1.22-alpine AS base

LABEL maintainer="Ambiente go"
LABEL description="Ambiente de desenvolvimento Go"

WORKDIR /app

# Instala dependências do sistema necessárias para o Go
RUN apk add --no-cache git build-base

# Copia os arquivos de dependência PRIMEIRO para aproveitar o cache do Docker
# Descomente e ajuste conforme o projeto usa go mod
# COPY go.mod go.sum ./
# RUN go mod download

# ==========================================
# ESTÁGIO 2: DESENVOLVIMENTO (DevContainer)
# ==========================================
FROM base AS dev

# Instala ferramentas úteis para desenvolvimento, como o reflex para hot-reload
RUN go install github.com/cespare/reflex@latest

EXPOSE 8080

# Comando para manter o container rodando e reagindo a mudanças (hot-reload)
# O código real será injetado via Bind Mount do compose.yaml
CMD ["sh", "-c", "reflex -r '\\.go$' -s -- sh -c 'go run ./'"]

# ==========================================
# ESTÁGIO 3: PRODUÇÃO (Deploy)
# ==========================================
FROM base AS prod

# Aqui copiamos o código para dentro da imagem
COPY . .

# Faz o build da aplicação
RUN go build -o main .

EXPOSE 8080

CMD ["./main"]
